### ISSUE TITLE
docs: add utility methods for import specifiers

### ISSUE THREAD

**Issue by mohab-sameh** [facebook/jscodeshift#614]
Document utility methods for import specifiers as described in #611.

**Comment by JakobJingleheimer** [comment]
Would be great to get some docs on these! The current documentation defines things with themselves, and there is no explanation on how to use them.

For instance, `filter(j.ImportSpecifier)`'s result do not contain the actual specifier. `filter(j.ImportDeclaration)`'s result does, but then re-composing a replacement is 🤯

**Comment by mohab-sameh** [comment]
@JakobJingleheimer agreed. I'll work on this soon. The current `AST Grammar` doc was supposed to be just an exhaustive reference of all the node type definitions rather than a usage guide. I'm wondering if that's even useful since it's not likely someone will need a list of node types in a doc format because:
1. they're already defined in the source code
2. tools like Codemod Studio and AST Explorer already allow users to know what node type a piece of code is

Should we have a doc with example usages of different node types instead? What's the ideal experience you want to see here?

**Comment by JakobJingleheimer** [comment]
Yeah, example usage would be great!

For instance when to filter by identifier vs importSpecifier vs importDeclaration etc and then how to build the return.
